File: 500+/560-subarray-sum-equals-k.cpp
int subarraySum(vector<int>& nums, int k) {
    int count = 0, psum = 0;
    unordered_map<int, int> hashmap;
    hashmap[0] = 1;
    for (const auto &num : nums) {
        psum += num;
        count += hashmap[psum-k];
        ++hashmap[psum];
    }
    return count;
}

File: python version/560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pre_sum = 0
        count = 0
        prefix_dict = collections.defaultdict(int)
        prefix_dict[0] = 1
        for num in nums:
            pre_sum += num
            count += prefix_dict[pre_sum-k]
            prefix_dict[pre_sum] += 1
        return count